### PR TITLE
Update aws sdk to v1.37.32

### LINKED ISSUE #229: Doesn't work with AWS SSO
I'm trying to make it work with AWS SSO but i'm getting `NoCredentialProviders`.

I used `aws configure sso` to create a profile on my local machine and i can manually `docker login` and pull images from ECR like this:
```
aws ecr get-login-password --region us-west-2 --profile my-profile | docker login --username AWS --password-stdin my-account-id.dkr.ecr.us-west-2.amazonaws.com

docker pull account-id.dkr.ecr.us-west-2.amazonaws.com/my-repo:latest
```

My `~/.aws/config` looks something like:
```
[profile my-profile]
sso_start_url = https://my-company.awsapps.com/start
sso_region = us-east-1
sso_account_id = my-account-id
sso_role_name = MyRoleName
region = us-west-2
output = json
```

Nothing was added to `~/.aws/credentials`.

I logged out docker, installed `docker-credential-helper-ecr` and set:

**~/.docker/config.json**
```
{
   "credsStore": "ecr-login"
}
```

and:
```
export AWS_PROFILE=my-profile
export AWS_REGION=us-west-2
```

When i:
```
docker pull account-id.dkr.ecr.us-west-2.amazonaws.com/repo:latest
```

I get:
```
no basic auth credentials
```

It works fine on another AWS account which is not SSO and i have my access keys set to `~/.aws/credentials`.

Does it work with AWS SSO somehow?

### PR DESCRIPTION
**Pull request by lachlancooper**
*Description of changes:*

Fixes #229

[aws-sdk-go v1.37.0](https://github.com/aws/aws-sdk-go/releases/tag/v1.37.0) includes support for the AWS Single Sign-On (SSO) credential provider.

This allows the use of AWS SSO directly, without requiring an intermediate `credential_process`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ecr-login/go.sum [hunk 1]
```diff
@@ -1,5 +1,7 @@
 github.com/aws/aws-sdk-go v1.36.0 h1:CscTrS+szX5iu34zk2bZrChnGO/GMtUYgMK1Xzs2hYo=
 github.com/aws/aws-sdk-go v1.36.0/go.mod h1:hcU610XS61/+aQV88ixoOzUoG7v3b31pl2zKMmprdro=
+github.com/aws/aws-sdk-go v1.37.0 h1:GzFnhOIsrGyQ69s7VgqtrG2BG8v7X7vwB3Xpbd/DBBk=
+github.com/aws/aws-sdk-go v1.37.0/go.mod h1:hcU610XS61/+aQV88ixoOzUoG7v3b31pl2zKMmprdro=
```
Comment (kzys) [inline ecr-login/go.sum:4]: Did you run `go mod tidy`? I think we should be able to remove 1.36.0 here.

Comment (lachlancooper) [reply ecr-login/go.sum]: Done in 09001a6049fe56beddcd98638846186d5f66c004

**Review by kzys — APPROVED** [review]

**Comment by samuelkarp** [comment]
@lachlancooper Thanks for opening this PR!  I haven't had the chance to try out AWS SSO yet; were you able to validate that this change makes it work properly?  If so, can you tell me a bit about how you tested it?

**Comment by lachlancooper** [comment]
> @lachlancooper Thanks for opening this PR! I haven't had the chance to try out AWS SSO yet; were you able to validate that this change makes it work properly? If so, can you tell me a bit about how you tested it?

Apologies, I should have included that information in the PR first up!

I validated this by successfully pulling a private ECR image, using an AWS profile properly configured for SSO:
```shell
# setup
$ aws sso login --profile test
<LOGIN REDACTED>
$ PATH="${PWD}/bin/local:${PATH}"

# build
$ git co update-aws-sdk-go
Switched to branch 'update-aws-sdk-go'
Your branch is up to date with 'lachlancooper/update-aws-sdk-go'.
$ make
git rev-parse --short=7 HEAD > GITCOMMIT_SHA
./scripts/build_binary.sh ./bin/local 0.5.0 09001a6
Built ecr-login
$ docker-credential-ecr-login -v
amazon-ecr-credential-helper
Version:    0.5.0
Git commit: 09001a6

# test
$ AWS_PROFILE=test docker pull <REDACTED>.dkr.ecr.us-east-1.amazonaws.com/<REDACTED>:1.13
1.13: Pulling from <REDACTED>
d6ff36c9ec48: Already exists
c958d65b3090: Already exists
edaf0a6b092f: Already exists
80931cf68816: Already exists
...
Status: Downloaded newer image for <REDACTED>.dkr.ecr.us-east-1.amazonaws.com/<REDACTED>:1.13
```

Compare that behaviour against a build from current `main`:
```shell
# build
$ git co main
Switched to branch 'main'
Your branch is up to date with 'origin/main'.
$ make
git rev-parse --short=7 HEAD > GITCOMMIT_SHA
./scripts/build_binary.sh ./bin/local 0.5.0 7cf3d3d
go: downloading github.com/aws/aws-sdk-go v1.36.0
Built ecr-login
$ PATH="${PWD}/bin/local:${PATH}"
$ docker-credential-ecr-login -v
amazon-ecr-credential-helper
Version:    0.5.0
Git commit: 7cf3d3d

# test
$ AWS_PROFILE=test docker pull <REDACTED>.dkr.ecr.us-east-1.amazonaws.com/<REDACTED>
Error response from daemon: Head https://<REDACTED>.dkr.ecr.us-east-1.amazonaws.com/v2/<REDACTED>: no basic auth credentials
```

Please let me know if there's any other information I can include here.

**Comment by DavidJacob** [comment]
@samuelkarp I can confirm this also works for me. I built @lachlancooper 's fork locally, logged into AWS SSO (which is the only AWS CLI access I have locally) and successfully pulled images from private repos.

**Comment by dancmeyers** [comment]
This is something that would also be really useful for us, as it's the last thing in the way of SSOifying our processes. 👍

**Comment by ejhayes** [comment]
This looks great -- any idea when this will be merged in?

**Comment by lachlancooper** [comment]
> I have the update and binaries on my fork of @lachlancooper here:
> [ejhayes/amazon-ecr-credential-helper@`0.5.0-1` (release)](https://github.com/ejhayes/amazon-ecr-credential-helper/releases/tag/0.5.0-1)

Thanks! I've pulled your commit into my branch.

**Comment by samuelkarp** [comment]
@lachlancooper Thanks for providing the testing details!

**Comment by mkantzer** [comment]
Now that this is merged, are there any plans on doing a release with this feature set included?

**Comment by eg-firebolt** [comment]
+1 Can we get a release? Looks like the last release on this repo was way back in Feb 2021 

https://github.com/awslabs/amazon-ecr-credential-helper/releases/tag/v0.5.0

**Pull request merged**